layer.py

```diff
@@ -417,7 +421,106 @@ def copy(self, target_path, copied_files, keep_existent=False):
             self._change_data_source(new_source)
         return copied_files
 
-    def _change_data_source(self, new_data_source):
+    def convert_to_gpkg(self, target_path):
+        """
+        Convert a layer to geopackage in the target path and adjust its datasource. If
+        a layer is already a geopackage, the dataset will merely be copied to the target
+        path.
+
+        :param layer: The layer to copy
+        :param target_path: A path to a folder into which the data will be copied
+        :param keep_existent: if True and target file already exists, keep it as it is
+        """
+
+        if (
+            not self.layer.type() == QgsMapLayer.VectorLayer
+            or not self.layer.dataProvider()
```

Comment:
```suggestion
            or not self.layer.isValid()
```
Can we change this and continue.